### ISSUE TITLE
Url::current behavior on error pages

### ISSUE THREAD

**Issue by thejahweh** [yiisoft/yii2#9072]
I have to use `Url:current()` in a layout which is great to do some fancy things for every page. My problem is just if an error occurs 404|403 or something else, then `Url:current()` returns the error `controller/action` url and not the current url.

Another thing which confused me is, if the error url is requested, it returns just a blank page. I would recommend to return a `404`, because the error page does not really exist in this way or should it?

**Comment by samdark** [comment]
`Url:current()` returns URL of current controller/action which, in case of error is `site/error`. It's kinda expected.

Didn't get the second part about blank page. Do you mean requesting `site/error`?

**Comment by klimov-paul** [comment]
> Didn't get the second part about blank page. Do you mean requesting site/error?

See https://github.com/yiisoft/yii2/blob/master/framework/web/ErrorAction.php#L78
Here if ther is no current exeption in `ErrorHandler` action just returns empty string.
Perhaps we should change this behavior to following:
```php
if (($exception = Yii::$app->getErrorHandler()->exception) === null) {
    $exception = new HttpException(404);
}
```

**Comment by samdark** [comment]
Ah, yes. Absolutely.

**Comment by klimov-paul** [comment]
Resolved by commit 984afd3
